chips/apollo3/src/iom.rs

```diff
@@ -1287,8 +1292,17 @@ impl<'a> SpiMaster<'a> for Iom<'a> {
             && transfered_bytes < 24
         {
             let idx = self.write_index.get();
-            let data =
-                u32::from_le_bytes(write_buffer[idx..(idx + 4)].try_into().unwrap_or([0; 4]));
+
+            let chunk = write_buffer[idx..].chunks(4).next().unwrap_or(&[]);
```

Comment:
```suggestion
            // Caution: This must handle "bytes remaining % 4 != 0" correctly.
            let chunk = write_buffer[idx..].chunks(4).next().unwrap_or(&[]);
```

Oh.. this is subtle, I had to read this a few times to figure out the problem. Nice catch.